README.md

```diff
@@ -229,6 +229,69 @@ Then run `now` and enjoy!
 
 Note: we recommend putting `.next` in `.npmignore` or `.gitignore`. Otherwise, use `files` or `now.files` to opt-into a whitelist of files you want to deploy (and obviously exclude `.next`)
 
+## Configuration
+
+While Next.js aims to work without any configuration, sometimes there is a need to add custom behaviour.
+You can define custom configuration in the configuration file called `next.config.js`, this file is stored in the root of the project. An example of a configuration
+looks like this:
+
+```javascript
+// next.config.js
+module.exports = {
+  cdn: true
+}
+```
+
+### Customizing webpack config
+
+Sometimes the user needs to have custom configuration for webpack to add a specific behaviour in the build process.
+An example of this is using `eslint-loader` to lint the files before compiling. This can be done by defining 
+`webpack` in the config.
+
+```javascript
+module.exports = {
+  webpack: (webpackConfig, { hotReload, dev }) => {
+    const newConfig = { ...webpackConfig }
+    newConfig.module.preLoaders.push({ test: /\.js$/, loader: 'eslint-loader' })
+    return newConfig
+  }
+}
+```
+
+As you can see you need to provide a function which has two parameters `webpackConfig`, which is the config used by Next.js and `options`, which
+contains `hotReload` (`true` if hotReload is on) and `dev` (`true` if dev environment). The config you return is the config used by Next.js.
+You can also return a `Promise` which will be resolved first.
+
+_NOTE: Use this option with care, because you can potentially break the existing webpack build configuration by using this option._
+
+These are some more examples:
+
+```javascript
+const I18nPlugin = require('i18n-webpack-plugin');
+
+module.exports = {
+  webpack: (webpackConfig, { hotReload, dev }) => {
+    const newConfig = { ...webpackConfig }
```

Comment:
Valid point. I added it as a best practice, because I don't like editing arguments directly.
I do however think this part adds an unnecessary complexity to the documentation, since here it is not necessary to copy first. It'll only raise question marks.

I'll remove it from the documentation.